patches/tModLoader/Terraria/ModLoader/TileLoader.cs

```diff
@@ -957,5 +957,34 @@ public static class TileLoader
 		public static bool UnlockChest(int i, int j, int type, ref short frameXAdjustment, ref int dustType, ref bool manual) {
 			return GetTile(type)?.UnlockChest(i, j, ref frameXAdjustment, ref dustType, ref manual) ?? false;
 		}
+
+		public static void RecountTiles(SceneMetrics metrics) {
+
+			// reset every tile count
+			metrics.HolyTileCount = metrics.EvilTileCount = metrics.BloodTileCount = metrics.SnowTileCount = metrics.JungleTileCount = metrics.MushroomTileCount = metrics.SandTileCount = 0;
+
+			for (int i = 0; i < TileCount; i++) {
```

Comment:
Should store the result of `metrics._tileCounts[i]` in a local variable, and `continue` the loop if it's 0.

Reply:
For my own curiosity, any reason to store it in a local variable rather than making a direct check? This is what I do in my most recent commit. Is this considered bad practice?

Reply:
Generally a good idea not to duplicate code (Terraria is a very poor example of this practice). Looking up an array repeatedly costs more than doing it once and saving the result.

Reply:
Also same goes for below, if you're going to check `TileID.Sets.HallowBiome[i] != 0` you may as well just do the multiplication and add directly.

Reply:
OH, I understand what you mean now. I somehow entirely overlooked that I use it for every line that it is adding to the counts even though that is the entire point, ahah.

I can fix that right away.